.github/actions/setup-self-hosted-action/action.yml

```diff
@@ -0,0 +1,50 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: 'Setup environment for self-hosted runners'
+description: 'Setup action to run jobs in a self-hosted runner'
+
+runs:
```

Comment:
It would be nice to have some optional inputs like:

```
inputs:
  requires-py-37:
    required: false
    default: 'true'
  requires-py-38:
    required: false
    default: 'true'
  requires-py-39:
    required: false
    default: 'true'
  requires-java-8:
    required: false
    default: 'true'
  requires-go-16:
    required: false
    default: 'true'
```

So that a given job can specify if it doesn't need some set of dependencies. I agree installing all of them by default is a good idea though.

Reply:
Hi Danny!
We can definitely add those inputs to this action so that further workflows can make use of them.